### PR TITLE
Update: Move elide metadata adapter to client pkg

### PR DESCRIPTION
**Pull request by jkusa**
Resolves #

## Proposed Changes

-

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/client/src/errors/navi-adapter-error.ts [hunk 1]
```diff
@@ -2,9 +2,8 @@
  * Copyright 2020, Yahoo Holdings Inc.
  * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
  */
-function isPresent<T>(t: T | undefined | null | void): t is T {
-  return t !== undefined && t !== null;
-}
+
+import { isPresent } from '../utils';
```
Comment (kevinhinterlong) [inline packages/client/src/errors/navi-adapter-error.ts:6]:
```suggestion
import { isPresent } from '../utils.js';
```

File: packages/client/src/plugins/elide-apollo-client.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    const host = config.configHost({ dataSourceName });
+    return new URL(host, `${document.location.protocol}${document.location.host}`).href;
+  }
+
+  function link() {
+    const linkOptions = { credentials: 'include' };
+    const httpLink = new HttpLink(linkOptions);
+
+    const contextLink = setContext((_, context) => {
+      context.headers = Object.assign(context.headers ?? {}, { clientId: 'UI' });
```
Comment (kevinhinterlong) [inline packages/client/src/plugins/elide-apollo-client.ts:21]: `// TODO: allow for clientId to be overridden`

File: packages/data/addon/adapters/metadata/elide.ts [hunk 3]
```diff
@@ -2,58 +2,12 @@
...
-        dataSourceName,
-        headers,
-      },
-      ...options,
-    };
-    return this.apollo.query(queryOptions);
+export default class NaviElideMetadataAdapter extends ElideMetadataAdapter {
+  static create(args: unknown) {
+    const owner = getOwner(args);
+    return new NaviElideMetadataAdapter(owner.lookup('service:client-injector'));
```
Comment (kevinhinterlong) [inline packages/data/addon/adapters/metadata/elide.ts:11]:
```suggestion
    return new this(owner.lookup('service:client-injector'));
```

File: packages/client/src/errors/navi-adapter-error.ts (hunk 1 above)
Comment (jkusa) [reply packages/client/src/errors/navi-adapter-error.ts:6]: I get this when I add the `.js`:
```
src/errors/navi-adapter-error.ts:6:27 - error TS2307: Cannot find module '../utils.js' or its corresponding type declarations.

6 import { isPresent } from '../utils.js';
                            ~~~~~~~~~~~~~


Found 1 error in src/errors/navi-adapter-error.ts:6
```

Comment (jkusa) [reply packages/client/src/errors/navi-adapter-error.ts]: Looks like you have to do `../utils/index.js` :/

**Review by kevinhinterlong — APPROVED** [review]

File: packages/client/src/plugins/elide/adapters/metadata.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import invariant from 'tiny-invariant';
+import { isPresent } from '../../../utils/index.js';
+import type NaviMetadataAdapter from '../../../adapters/metadata/interface';
+import type { MetadataOptions } from '../../../adapters/metadata/interface';
+import { QueryOptions } from '@apollo/client/core';
+import NativeWithCreate, { Config } from '../../../models/native-with-create.js';
+import type { Injector } from '../../../models/native-with-create.js';
+import type { ApolloClient } from '@apollo/client/core';
+import getClient from '../../elide-apollo-client.js';
+import type { ClientConfig } from '../../../config/datasources';
```
Comment (kevinhinterlong) [inline packages/client/src/plugins/elide/adapters/metadata.ts:15]:
```suggestion
import queries from '../../../gql/metadata-queries.js';
import invariant from 'tiny-invariant';
import { isPresent } from '../../../utils/index.js';
import type NaviMetadataAdapter from '../../../adapters/metadata/interface.js';
import type { MetadataOptions } from '../../../adapters/metadata/interface.js';
import { QueryOptions } from '@apollo/client/core';
import NativeWithCreate, { Config } from '../../../models/native-with-create.js';
import type { Injector } from '../../../models/native-with-create.js';
import type { ApolloClient } from '@apollo/client/core';
import getClient from '../../elide-apollo-client.js';
import type { ClientConfig } from '../../../config/datasources.js';
```

**Pull request merged**